### ISSUE TITLE
Allow building for 32-bit arch on 64-bit Linux

### ISSUE THREAD

**Issue by mratsim** [status-im/nim-beacon-chain#1723]
Currently we can't build NBC for 32-bit on 64-bit Linux due to miniupnp and libnatpmp in https://github.com/status-im/nim-nat-traversal.

This can be tested in Github action: https://github.com/status-im/nim-beacon-chain/pull/1693#issuecomment-696623185

**Comment by stefantalpalaru** [comment]
Did you try adding `-m32` to CFLAGS?

**Comment by stefantalpalaru** [comment]
You probably already have that set by the CI runner. Just bump "vendor/nimbus-build-system" to get the latest fix (the "libnatpmp.a" target was ignoring CFLAGS from the environment).

**Comment by mratsim** [comment]
It's set there: https://github.com/status-im/nim-beacon-chain/pull/1693/files#diff-e9f950f17198d3d5e3122a44230a09b9R114-R121

```yaml

      - name: Install build dependencies (Linux i386)
        if: runner.os == 'Linux' && matrix.target.cpu == 'i386'
        run: |
          sudo dpkg --add-architecture i386
          sudo apt-fast update -qq
          sudo DEBIAN_FRONTEND='noninteractive' apt-fast install \
            --no-install-recommends -yq gcc-multilib g++-multilib \
            libssl-dev:i386 libpcre3-dev:i386
          mkdir -p external/bin
          cat << EOF > external/bin/gcc
          #!/bin/bash
          exec $(which gcc) -m32 "\$@"
          EOF
          cat << EOF > external/bin/g++
          #!/bin/bash
          exec $(which g++) -m32 "\$@"
          EOF
          chmod 755 external/bin/gcc external/bin/g++
          echo '::add-path::${{ github.workspace }}/external/bin'
```

**Comment by stefantalpalaru** [comment]
That looks like an ugly hack... Don't you need to do the same for `ld`?

**Comment by mratsim** [comment]
You gotta do what you gotta do :man_shrugging:, it works fine in nim-blscurve https://github.com/status-im/nim-blscurve/blob/eed30c06231438b87fe25127326e40cb2bf412e3/.github/workflows/ci.yml#L59-L77

**Comment by mratsim** [comment]
Solved, #1693 is working
